Q2ShawnCampbell.md

```diff
@@ -0,0 +1,148 @@
+# Quiz #2
+
+## Instructions
+
+1. Fork this repo
+2. Clone your fork
+3. Fill in your answers by writing in the appropriate area, or placing an 'x' in
+the square brackets (for multiple-choice questions).
+4. Add/Commit/Push your changes to Github.
+5. Open a pull request.
+
+## CSS
+
+### Question #1
+
+Describe the purpose of a clearfix in CSS, and give an example of how to do it.
+
+Your Answer:
+```The clearfix is a CSS method used to make it easier to format a group of
+images.
+```
+
+### Question #2
+
+What does the following selector do?  `ul.dropdown > li`?
+
+Select 1:
+```
+[] Selects all li's which are directly inside a ul of class dropdown (children)
+[X] Selects all li's which are anywhere inside a ul of class dropdown (any descendant)
+[] Selects all ul's of class dropdown, as well as the children elements that are li's
+[] Selects all ul's of class dropdown, only if their children are exclusively li's
+```
+
+## Scope/Context/Closures
+
+### Question #3
+
+Describe the rules of scope in JavaScript.
+
+Your Answer:
+```In Javascript, scope refers to the limits of using types of variables. Global
+variables, those not defined only by one function, can be used in any function.
+Local variables can only be used in specific functions that they were
+defined for, and created within.
+```
+
+
+### Question #4
+
+Define an object and store it in a variable `pizza`. The object should have 2
+properties: a temperature (set to 70), and a method called `bake`. When called,
+this method should set the pizza's temperature to be 300. Note: you may not use
+the variable pizza inside your method.
+
+Your Answer:
+```js
+<var pizza = {
+  temperature: 70;
+  function bake() {
+    console.log("the oven has been set to 300")
+}
+}
+```
+
+## Callbacks
+
+### Question #5
+
+**Define a function called `doSomething`. It should take one argument, called
+`thingToDo`. When called, `doSomething` should invoke the function given as an
+argument. Finally, demonstrate calling `doSomething` with a function.**
+
+Your Answer:
+```js
+function doSomething() {
+  thingToDo = value * 10
+  doSomething.thingToDo(5)
+}
+```
+
```

Comment:
It looks like you're treating `thingToDo` as an object method. It is supposed to be an argument passed into `doSomething` that represents a function. In the `doSomething` function definition, you need to call `thingToDo` (i.e., `thingToDo( )`.

Let's talk about this during our 1:1.